### PR TITLE
Test terraform cloudflare resources logging

### PR DESCRIPTION
**Pull request by esron**
Relates to: [APPSRE-8315](https://issues.redhat.com/browse/APPSRE-8315)

Adds tests to the validation logging in `terraform-cloudflare-resources`.

### CODE REVIEW

File: reconcile/test/test_terraform_cloudflare_resources.py [hunk 1]
```diff
@@ -104,3 +114,165 @@ def external_resources(provisioner_config):
...
+
+
+@pytest.fixture
+def mock_cloudflare_accounts(mocker):
+    from reconcile.gql_definitions.terraform_cloudflare_resources.terraform_cloudflare_accounts import (
+        CloudflareAccountV1 as CFAccountV1,
+    )
+    from reconcile.gql_definitions.terraform_cloudflare_resources.terraform_cloudflare_accounts import (
+        TerraformCloudflareAccountsQueryData,
+    )
```
Comment (hemslo) [inline reconcile/test/test_terraform_cloudflare_resources.py:140]: Any reason to reimport these?

File: reconcile/test/test_terraform_cloudflare_resources.py [hunk 2]
```diff
@@ -104,3 +114,165 @@ def external_resources(provisioner_config):
...
+    mocked_cloudflare_resources.query.return_value = external_resources
+
+
+def test_cloudflare_accounts_validation(
+    mocker, caplog, mock_gql, mock_vault_secret, mock_cloudflare_resources
+):
+    from reconcile.gql_definitions.terraform_cloudflare_resources.terraform_cloudflare_accounts import (
+        TerraformCloudflareAccountsQueryData,
+    )
```
Comment (hemslo) [inline reconcile/test/test_terraform_cloudflare_resources.py:202]: why not import this together at top?

**Review by hemslo — APPROVED** [review]

File: reconcile/test/test_terraform_cloudflare_resources.py (hunk 1 above)
Comment (esron) [reply reconcile/test/test_terraform_cloudflare_resources.py:140]: Yes. They have different paths. The ones on the top of the file are from `terraform_cloudflare_resources` they are slightly different.

File: reconcile/test/test_terraform_cloudflare_resources.py (hunk 2 above)
Comment (esron) [reply reconcile/test/test_terraform_cloudflare_resources.py]: I've put all the imports on the top. 👍🏽

**Pull request merged**